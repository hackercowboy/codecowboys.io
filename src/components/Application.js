import React from 'react';
import PropTypes from 'prop-types';
import { injectIntl } from 'react-intl';
import Image from './Image';
import './Application.scss';

function Application({
  title, teaser, image, link, intl,
}) {
  return (
    <div className="application">
      <h4>{title}</h4>
      <div className="application-wrapper">
        <div className="application-teaser">
          <p>
            {teaser}
            {' '}
            <a href={link}>
              {intl.formatMessage({ id: 'common.more' })}
              ...
            </a>
          </p>
        </div>
        <a className="application-image" href={intl.formatMessage({ id: 'letter.link' })}>
          <Image src={image} width={100} height={100} alt={title} />
        </a>
      </div>
    </div>
  );
}

Application.propTypes = {
  title: PropTypes.string.isRequired,
  teaser: PropTypes.string.isRequired,
  link: PropTypes.string.isRequired,
  image: PropTypes.string.isRequired,
  intl: PropTypes.shape({
    formatMessage: PropTypes.func,
  }).isRequired,
};

export default injectIntl(Application);
