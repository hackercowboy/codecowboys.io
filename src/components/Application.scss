.application {
  margin-bottom: 50px;

  h4 {
    margin: 0;
  }
}

.application-wrapper {
  align-items: center;
  display: flex;
}

.application-teaser {
  flex-grow: 1;
}

.application-image {
  margin-left: 10px;
  width: 100px;
}
