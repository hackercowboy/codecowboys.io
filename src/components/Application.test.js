import React from 'react';
import { shallow } from 'enzyme';

import Application from './Application';

describe('<Application/>', () => {
  it('should render', () => {
    const wrapper = shallow(<Application />);
    expect(wrapper.debug()).toMatchSnapshot();
  });
});