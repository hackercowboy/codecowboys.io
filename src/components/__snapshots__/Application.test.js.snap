// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Application/> should render 1`] = `"<Application intl={{...}} />"`;
